File: src/minecraft/net/machinemuse/powersuits/powermodule/misc/BinocularsModule.scala
package net.machinemuse.powersuits.powermodule.misc

import net.machinemuse.powersuits.powermodule.PowerModuleBase
import net.machinemuse.utils.{MusePlayerUtils, MuseItemUtils, MuseCommonStrings}
import net.machinemuse.powersuits.item.ItemComponent
import net.minecraft.item.ItemStack
import net.minecraft.entity.player.EntityPlayer
import net.machinemuse.powersuits.control.KeybindKeyHandler
import cpw.mods.fml.common.FMLCommonHandler
import net.machinemuse.api.{ModuleManager, IModularItem}
import net.machinemuse.api.moduletrigger.IPlayerTickModule

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 1:08 AM, 4/24/13
 */
class BinocularsModule(list: java.util.List[IModularItem]) extends PowerModuleBase(list) with IPlayerTickModule {
  val FOV_MULTIPLIER = "Field of View"
  addInstallCost(MuseItemUtils.copyAndResize(ItemComponent.laserHologram, 1))
  addBaseProperty(FOV_MULTIPLIER, 0.5)
  addTradeoffProperty("FOV multiplier", FOV_MULTIPLIER, 9.5, "%")

  override def getCategory: String = MuseCommonStrings.CATEGORY_VISION

  override def getName: String = "Binoculars"

  override def getDescription: String = "With the problems that have been plaguing Optifine lately, you've decided to take that Zoom ability into your own hands."

  override def getTextureFile: String = "binoculars"

  override def onPlayerTickActive(player: EntityPlayer, item: ItemStack) {
    if (FMLCommonHandler.instance().getEffectiveSide.isClient) {
      val fov = KeybindKeyHandler.zoom.pressed match {
        case true => ModuleManager.computeModularProperty(item, FOV_MULTIPLIER).toFloat
        case false => 0.1f
      }
      MusePlayerUtils.setFOVMult(player, fov)
    }

  }

  override def onPlayerTickInactive(player: EntityPlayer, item: ItemStack) {
  }
}

File: src/main/java/net/machinemuse/powersuits/common/powermodule/misc/BinocularsModule.java
package net.machinemuse.powersuits.common.powermodule.misc;

import net.machinemuse.api.IModularItem;
import net.machinemuse.api.moduletrigger.IToggleableModule;
import net.machinemuse.powersuits.client.events.MuseIcon;
import net.machinemuse.powersuits.common.items.ItemComponent;
import net.machinemuse.powersuits.common.powermodule.PowerModuleBase;
import net.machinemuse.utils.MuseCommonStrings;
import net.machinemuse.utils.MuseItemUtils;
import net.minecraft.client.renderer.texture.TextureAtlasSprite;
import net.minecraft.item.ItemStack;

import java.util.List;

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 1:08 AM, 4/24/13
 *
 * Ported to Java by lehjr on 10/11/16.
 */
public class BinocularsModule extends PowerModuleBase implements IToggleableModule {
    public static final String BINOCULARS_MODULE = "Binoculars";
    public static final String FOV_MULTIPLIER = "Field of View";

    public BinocularsModule(List<IModularItem> validItems) {
        super(validItems);
        addInstallCost(MuseItemUtils.copyAndResize(ItemComponent.laserHologram, 1));
        addBaseProperty(BinocularsModule.FOV_MULTIPLIER, 0.5);
        addTradeoffProperty("FOV multiplier", BinocularsModule.FOV_MULTIPLIER, 9.5, "%");
    }

    @Override
    public String getCategory() {
        return MuseCommonStrings.CATEGORY_VISION;
    }

    @Override
    public String getDataName() {
        return BinocularsModule.BINOCULARS_MODULE;
    }

    @Override
    public String getUnlocalizedName() {
        return "binoculars";
    }

    @Override
    public TextureAtlasSprite getIcon(ItemStack item) {
        return MuseIcon.binoculars;
    }
}